src/v/cluster/scheduling/constraints.cc

```diff
@@ -168,13 +168,15 @@ soft_constraint_evaluator distinct_rack(
             for (auto [node_id, shard] : _replicas) {
                 auto rack = _state.get_rack_id(node_id);
                 if (!rack.has_value()) {
-                    return 1;
+                    return soft_constraint_evaluator::max_score
```

Comment:
good idea. is there a 'hard' constraint on this placement? thinking about how to document it. cc: @coral-waters

Reply:
there is no hard constraint (it will cause problems when we will have fewer racks than replicas)
so for now it's soft constraint